00_Base/src/model/DTO/admin/AdminLocationDTO.ts

```diff
@@ -0,0 +1,153 @@
+import { GeoLocation } from '../../GeoLocation';
+import {
+  IsArray,
+  IsBoolean,
+  IsNotEmpty,
+  IsNumber,
+  IsObject,
+  IsString,
+  MaxLength,
+  ValidateNested,
+} from 'class-validator';
+import { Optional } from '../../../util/decorators/optional';
+import { Type } from 'class-transformer';
+
+const LOCATION_INVALID_MSG =
+  'New locations must have party_id, country_code, name, address, city, state, postal_code, and country.';
+const EVSE_INVALID_MSG = 'EVSEs must have id and station_id.';
+const CONNECTOR_INVALID_MSG = 'Connectors must have an id.';
+
+// TODO add remaining OCPI-specific properties
+export class AdminLocationDTO {
+  // id is the Citrine Location database id, not the OcpiLocation's database id
+  @IsNumber()
+  @Optional()
+  id?: number;
+
+  @IsString()
+  @MaxLength(2)
+  @Optional()
+  country_code!: string;
+
+  @IsString()
+  @MaxLength(3)
+  @Optional()
```

Comment:
Why the optional annotation for party id, country code, and id?

Shouldn't these be required?

Reply:
they're optional if the AdminLocationDTO is used in the "update" form rather than the "create" form :( i didn't play around much with the annotations though to see if there was a way to tell it "ignore the validation under these circumstances" -- let me see if there's a way to get that going, then i can remove that validation method 🤔

Reply:
In that case, maybe we should define a `CreateLocationRequest` and an `UpdateLocationRequest`? 

I guess the only issue there is that there might be a lot of redundancy.

Reply:
aha, figured it out! you can add validation groups 😄 validation groups now added and they're properly preventing empty values only on create